test/com/google/enterprise/adaptor/DocumentHandlerTest.java

```diff
@@ -2481,6 +2487,545 @@ public void testPercentEncoding() {
                  + "%2A%28%29%5B%5D%7B%7D%C3%AB%01", encoded);
   }
 
+  // tests for hostsSkipCertCheck
+
+  // server.fullAccessHosts=NOT_OUR_IP_ADDRESS
+  // server.skipCertCheckHosts=NOT_OUR_IP_ADDRESS
+  // adaptor trusts to CN=localhost
+  // requestor ip=127.0.0.3
+  // requestor CN=localhostbad
+  // result: 403, no headers
+  @Test
+  public void testSkipCertCheckBadCNBadIP() throws Exception {
+    MockHttpsExchange httpsEx = new MockHttpsExchange(ex, new MockSslSession(
+        new X500Principal("CN=localhostbad, OU=Unknown, O=Unknown, C=Unknown")));
+
+    final String key = "testing key";
+    List<MetadataTransform> transforms
+        = new LinkedList<MetadataTransform>();
+    transforms.add(new MetadataTransform() {
+      @Override
+      public void transform(Metadata metadata, Map<String, String> params) {
+        metadata.set(key, metadata.getOneValue(key).toUpperCase());
+      }
+    });
+    MetadataTransformPipeline transform = new MetadataTransformPipeline(transforms,
+        Arrays.asList("t1"));
+
+    UserPrivateMockAdaptor adaptor = new UserPrivateMockAdaptor() {
+      @Override
+      public void getDocContent(Request request, Response response)
+          throws IOException, InterruptedException {
+        response.addMetadata(key, "testing value");
+        super.getDocContent(request, response);
+      }
+    };
+    DocumentHandler handler = createHandlerBuilder()
+        .setAdaptor(adaptor)
+        .setAuthzAuthority(adaptor)
+        .setFullAccessHosts(new String[] {NOT_OUR_IP_ADDRESS})
+        .setSkipCertHosts(new String[] {NOT_OUR_IP_ADDRESS})
+        .setMetadataTransform(transform)
+        .build();
+    mockAdaptor.documentBytes = new byte[] {1, 2, 3};
+
+    handler.handle(httpsEx);
+
+    // response code 403 - adaptor doesn't trust to CN=badlocalhost
```

Comment:
whatis "localhostbad" ?  is it the same as "badlocalhost"?  

Can we say "untrusted" rather than "badlocalhost"?